packages/devtools_app/lib/src/code_size/file_import_container.dart

```diff
@@ -0,0 +1,222 @@
+                      _buildImportedFileDisplay(context),
+                  ],
+                ),
+                if (widget.actionText != null && widget.onAction != null)
+                  _buildAnalyzeButton(widget.actionText),
+              ],
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Column _buildImportedFileDisplay(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const SizedBox(height: defaultSpacing),
+        Text(
+          'Imported File:',
+          textAlign: TextAlign.center,
+        ),
+        Text(
+          importedFile,
+          style: TextStyle(color: Theme.of(context).textTheme.headline1.color),
+          textAlign: TextAlign.center,
+        ),
+      ],
+    );
+  }
+
+  Row _buildImportButton() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        OutlineButton(
+          onPressed: () {
+            // TODO(peterdjlee): Prompt file picker to choose a file.
+            setState(() {
+              if (widget.importNewFile) {
+                importedFile =
+                    '$current/lib/src/code_size/stub_data/old_v8.json';
+              } else {
+                importedFile =
+                    '$current/lib/src/code_size/stub_data/new_v8.json';
+              }
+            });
+
+            if (widget.onFileSelected != null)
+              widget.onFileSelected(importedFile);
+          },
+          child: const MaterialIconLabel(Icons.file_upload, 'Import File'),
+        ),
+      ],
+    );
+  }
+
+  Column _buildAnalyzeButton(String title) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const SizedBox(height: defaultSpacing),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            RaisedButton(
+              onPressed: importedFile != null
+                  ? () => widget.onAction(importedFile)
+                  : null,
+              child: MaterialIconLabel(
+                Icons.highlight,
+                title,
+              ),
+            ),
+          ],
+        ),
+      ],
+    );
+  }
+}
+
+class DualFileImportContainer extends StatefulWidget {
+  const DualFileImportContainer({
+    @required this.actionText,
+    @required this.onAction,
+    Key key,
+  }) : super(key: key);
+
+  /// The title of the action button.
+  final String actionText;
+
+  final Function(String oldFilePath, String newFilePath) onAction;
+
+  @override
+  _DualFileImportContainerState createState() =>
+      _DualFileImportContainerState();
+}
+
+class _DualFileImportContainerState extends State<DualFileImportContainer> {
+  String pathToOldFile;
+  String pathToNewFile;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: FileImportContainer(
+            title: 'Old',
```

Comment:
add parameters to DualFileImportContainer `firstFileTitle` `secondFileTitle` to make this configurable, because we may need a dual file picker for something else in the future that is not specifically "old" and "new"